### PR TITLE
Added flask-cors dependency and CORS_ENABLE env var to enable

### PR DESCRIPTION
**Pull request by Jacq**
### Added flask-cors dependency and CORS_ENABLE env var to enable

Added the CORS_ENABLE os env flag to activate the CORS for all routes, I did not find an app.config setup that could be used instead, to avoid using a os env. 

I have no much experience with Flask, app.config['CORS_HEADERS'] is currently not set, could this be set by the general Flask app.config?, I tried to do that adding this general code for configuration in __init__.py after the Flask app init  'app = Flask(__name__)'
``` 
if test_config is None:
        # load the instance config, if it exists, when not testing
        app.config.from_pyfile('config.py', silent=True)
    else:
        # load the test config if passed in
        app.config.from_mapping(test_config)
```  
, but then the cli tests keep failing due to the Mock failing:
`AttributeError: <class 'flask.app.Flask'> does not have the attribute 'config'`
, I think the Mock for the Flask app is missing the config attribute only for the cli tests.

### Added also a Dockerfile and docker-compose.yml to setup environment. 
Added dockerfile from python:3 base, in the future this could be use for production, but then an alpine smaller image should be used and a more production ready WGSI server?

Only prod dependencies are installed, there is a long discussion about the best way to integrate poetry and Dockerfile to benefit from cache and keep dev and prod environments similar: https://github.com/python-poetry/poetry/discussions/1879
I do not have experience with poetry so I have just created a simple Dockerfile with no dev dependencies so it could be used for production. For testing 'poetry install' should be run from the container to have "poetry poe test" available.

Cheers
Jacq

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
+FROM python:3
```
Comment (amotl) [inline Dockerfile:1]: Will bring the base image size down from 921MB to 126MB.
```suggestion
FROM python:3-slim
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+RUN poetry config virtualenvs.create false \
+  && poetry install --no-interaction --no-ansi --only main
+
+# TODO move this after poetry install but then pyprojet.toml packages entry cannot find wireviz_web
+COPY . .
+  
+ENV FLASK_APP=wireviz_web
+
+EXPOSE 3005 5000
```
Comment (amotl) [inline Dockerfile:24]: What's on port 5000? Can you remove this port otherwise?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+services:
+  wireviz-web:
+    build: .
+    volumes:
+     - .:/usr/src/app
+    environment:
+      - CORS_ENABLE=true
+    ports: 
+     - 3005:3005
+     - 5000:5000
```
Comment (amotl) [inline docker-compose.yml:12]: Dito. Can you remove port 5000 here?

**Review by amotl — CHANGES_REQUESTED** [review]
Dear Jacq,

thanks a stack for your excellent contribution, it is very much appreciated. Can I humbly ask you to adjust three details on this patch?

- Check my inline comments.
- Format the code with `poetry run poe format`.
- Split the patch into two commits, one for activating CORS, and another one for adding the Docker infrastructure.

I will be happy to run a new release right after merging.

Thank you in advance and with kind regards,
Andreas.

File: Dockerfile (hunk 2 above)
Comment (Jacq) [reply Dockerfile:24]: It was from some testing during the Dockerfile construction, it is the default Flask port, I will remove it.

**Comment by amotl** [comment]
Closing this in favor of #9 and #10. Thank you again!

**Pull request closed without merging**